src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1484,12 +1484,6 @@ void TorrentHandle::toggleFirstLastPiecePriority()
 void TorrentHandle::pause()
 {
     if (m_startupState != Started) return;
```

Comment:
Seems to handle the second case we just need to set `m_startupState` to `Started` under this condition.

Reply:
```
 void TorrentHandle::pause()
 {
-    if (m_startupState != Started) return;
     if (isPaused()) return;
 
+    m_startupState = Started;
     setAutoManaged(false);
     m_nativeHandle.pause();
```
or
```
 void TorrentHandle::pause()
 {
-    if (m_startupState != Started) return;
+    if (m_startupState != Started) {
+        if (isChecking())
+            m_startupState = Started;
+        else
+            return;
+    }
     if (isPaused()) return;
 
     setAutoManaged(false);
```

Reply:
```c++
if (m_startupState != Started)
    m_startupState != Started;
```

But, seems, you need to move it to the end of the method.

Reply:
Done, thanks.

Reply:
:+1:
Now can you cause "checking on startup" case for some paused torrent to test it? Probably you need to trigger rechecking of some paused torrent and then close the app before it's done.

Reply:
When i do it to a seeding torrent, recheck->close qBt before it finishes->start qBt and pause rechecking torrent->resume/force resume
After the recheck finishes, the torrent becomes paused. Shouldn't it go back to seeding?

Reply:
>Shouldn't it go back to seeding?

Sure.
>recheck->close qBt before it finishes

You don't pause it before recheck? Then it is a different kind of (related) issue.

Reply:
Rechecking paused torrent doesn't have any problems, works as expected.

Reply:
>Rechecking paused torrent 

We do talk about ability to pause (and then resume) checking torrents, don't it? So starting state is "checking" torrent.
Really we have 4 starting (sub)states that should be tested:
1. Forced checking torrent (torrent was resumed when user triggers force recheck);
2. Forced checking paused torrent  (torrent was paused when user triggers force recheck);
3. Startup checking torrent (torrent is added/restored as resumed, e.g. it was resumed when app is closed last time);
4. Startup checking paused torrent (torrent is added/restored as paused, e.g. it was paused when app is closed last time).

In any of these cases the torrent should remain resumed after checking has done when user (pauses and then) resumes it explicitly.

@thalieht, do you understand what I'm talking about? If so can you please test and report all of these cases above?